internal/resources/grafana/resource_sso_settings_test.go

```diff
@@ -0,0 +1,292 @@
+					resource.TestCheckResourceAttr("grafana_sso_settings.second_sso_settings", "oauth2_settings.0.client_id", "second_gitlab_client_id"),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+			{
+				ResourceName:            "grafana_sso_settings.first_sso_settings",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"oauth2_settings.0.client_secret"},
+			},
+			{
+				ResourceName:            "grafana_sso_settings.second_sso_settings",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"oauth2_settings.0.client_secret"},
+			},
+			{
+				RefreshState: true,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("grafana_sso_settings.first_sso_settings", "provider_name", provider),
+					resource.TestCheckResourceAttr("grafana_sso_settings.first_sso_settings", "oauth2_settings.#", "1"),
+					resource.TestCheckResourceAttr("grafana_sso_settings.second_sso_settings", "provider_name", provider),
+					resource.TestCheckResourceAttr("grafana_sso_settings.second_sso_settings", "oauth2_settings.#", "1"),
+					func(s *terraform.State) error {
+						first, err := getPrimaryInstanceState(s, "grafana_sso_settings.first_sso_settings")
+						if err != nil {
+							return err
+						}
+
+						second, err := getPrimaryInstanceState(s, "grafana_sso_settings.second_sso_settings")
+						if err != nil {
+							return err
+						}
+
+						firstClientID, ok := first.Attributes["oauth2_settings.0.client_id"]
+						if !ok {
+							return fmt.Errorf("client_id not found in settings")
+						}
+
+						secondClientID, ok := second.Attributes["oauth2_settings.0.client_id"]
+						if !ok {
+							return fmt.Errorf("client_id not found in settings")
+						}
+
+						if firstClientID != secondClientID {
+							return fmt.Errorf("client_id is not the same: %s vs. %s", firstClientID, secondClientID)
+						}
+
+						return nil
+					},
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestSSOSettings_resourceWithNoSettings(t *testing.T) {
+	testutils.CheckOSSTestsEnabled(t, ">=10.4.0")
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProviderFactories: testutils.ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config:      testConfigWithNoSettings,
+				ExpectError: regexp.MustCompile("Insufficient oauth2_settings blocks"),
+			},
+		},
+	})
+}
+
+func TestSSOSettings_resourceWithEmptySettings(t *testing.T) {
+	testutils.CheckOSSTestsEnabled(t, ">=10.4.0")
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProviderFactories: testutils.ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config:      testConfigWithEmptySettings,
+				ExpectError: regexp.MustCompile("Missing required argument"),
+			},
+		},
+	})
+}
+
+func TestSSOSettings_resourceWithManySettings(t *testing.T) {
+	testutils.CheckOSSTestsEnabled(t, ">=10.4.0")
```

Comment:
For these tests that don't actually end up calling the Grafana instance, you can remove the semver condition. It fails before even calling the API so it will work with any Grafana version

Reply:
Done.